docs/proposals/network-policy.md

```diff
@@ -0,0 +1,335 @@
+type NetworkPolicyIngressRule struct {
+	// List of ports which should be made accessible on the pods selected for this rule. 
+	// Each item in this list is combined using a logical OR.  
+	// If this field is not provided, this rule matches all ports (traffic not restricted by port). 
+	// If this field is empty, this rule matches no ports (no traffic matches).
+	// If this field is present and contains at least one item, then this rule allows traffic 
+	// only if the traffic matches at least one port in the ports list. 
+	Ports *[]NetworkPolicyPort `json:"ports,omitempty"`
+	
+	// List of sources which should be able to access the pods selected for this rule.
+	// Items in this list are combined using a logical OR operation.
+	// If this field is not provided, this rule matches all sources (traffic not restricted by source).
+	// If this field is empty, this rule matches no sources (no traffic matches). 
+	// If this field is present and contains at least on item, this rule allows traffic only if the 
+	// traffic matches at least one item in the from list. 
+	From *[]NetworkPolicyPeer `json:"from,omitempty"`
+}
+
+type NetworkPolicyPort struct {
+	// Optional.  The protocol (TCP or UDP) which traffic must match.
+	// If not specified, this field defaults to TCP. 
+	Protocol *api.Protocol `json:"protocol,omitempty"`
+	
+	// If specified, the port on the given protocol.  This can 
+	// either be a numerical or named port.  If this field is not provided,
+	// this matches all port names and numbers.
+	// If present, only traffic on the specified protocol AND port
+	// will be matched.
+	Port *intstr.IntOrString `json:"port,omitempty"`
+}
+
+type NetworkPolicyPeer struct {
+	// Exactly one of the following must be specified.
+
+	// This is a label selector which selects Pods in this namespace.
+	// This field follows standard unversioned.LabelSelector semantics.
+	// If not provided, this selector selects no pods.
+	// If present but empty, this selector selects all pods in this namespace.
+	PodSelector *unversioned.LabelSelector `json:"podSelector,omitempty"`
+	
+	// Selects Namespaces using cluster scoped-labels.  This 
+	// matches all pods in all namespaces selected by this label selector. 
+	// This field follows standard unversioned.LabelSelector semantics.
+	// If omited, this selector selects no namespaces.
+	// If present but empty, this selector selects all namespaces.
+	NamespaceSelector *unversioned.LabelSelector `json:"namespaceSelector,omitempty"`
+}
+```
+
+### Behavior
+
+The following pseudo-code attempts to define when traffic is allowed to a given pod when using this API.
+
+```python
+def is_traffic_allowed(traffic, pod):
+  """
+  Returns True if traffic is allowed to this pod, False otherwise.
+  """
+  if not pod.Namespace.Spec.NetworkPolicy.Ingress.Isolation:
+    # If ingress isolation is disabled on the Namespace, all traffic is allowed.
+    return True 
```

Comment:
I thought we agreed this should say to return `clusterDefault(traffic, pod)` instead of `True`